core/utils/src/transaction/runner.rs

```diff
@@ -285,119 +284,4 @@ impl<C: NodeComponents> TransactionRunner<C> {
             }
         }
     }
-
-    async fn ensure_nonce_used(
```

Comment:
@matthias-wright does this make sense to you? If we're going to attempt to use the same nonce whenever `self.nonce_state.get_base().await < tx_nonce`, this whole thing seemed unnecessary, but please sanity check 🙏